src/OrchardCore.Modules/OrchardCore.ContentFields/Settings/UserPickerFieldSettingsDriver.cs

```diff
@@ -43,14 +51,17 @@ public override async Task<IDisplayResult> UpdateAsync(ContentPartFieldDefinitio
                 settings.Hint = model.Hint;
                 settings.Required = model.Required;
                 settings.Multiple = model.Multiple;
-                settings.DisplayAllUsers = model.DisplayAllUsers;
-                if (settings.DisplayAllUsers)
+                var selectedRoles = model.Roles.Where(x => x.IsSelected).Select(x => x.Role).ToArray();
+
+                if (model.DisplayAllUsers || selectedRoles.Length == 0)
                 {
-                    settings.DisplayedRoles = Array.Empty<String>();
+                    // No selected role should have the same effect as display all users
+                    settings.DisplayedRoles = Array.Empty<string>();
+                    settings.DisplayAllUsers = true;
                 }
                 else
                 {
-                    settings.DisplayedRoles = model.Roles.Where(x => x.IsSelected).Select(x => x.Role).ToArray();
+                    settings.DisplayedRoles = selectedRoles;
```

Comment:
settings.DisplayAllUsers = false;

Reply:
Because here it is a UserPickerFieldSettings which always defaults to true from a new instance.